core/controller/src/main/scala/whisk/core/controller/Controller.scala

```diff
@@ -185,7 +187,49 @@ object Controller {
     }
 
     def main(args: Array[String]): Unit = {
-        implicit val actorSystem = ActorSystem("controller-actor-system")
+
+        // read controller parameters from env variables
+        val port = sys.env("AKKA_CLUSTER_PORT")
+        val bindPort = sys.env("AKKA_CLUSTER_BIND_PORT")
+        val hostname = sys.env("AKKA_CLUSTER_HOST")
+        val bindHost = InetAddress.getLocalHost.getHostAddress
+
+        val conf = s"""
+                      include "logging"
+                      akka {
+                         actor {
+                           provider = cluster
+                         }
+                         remote {
+                           log-remote-lifecycle-events = off
+                           netty.tcp {
```

Comment:
`hostname` and `port` should be different from `bind-hostname` and `bind-port` when running in Docker. The latter 2 should bind to Docker's private IP Address and port, and the first 2 should bind to host's IP and port.